### DISCUSSION TITLE
Error while compiling using internal DAC

### DISCUSSION THREAD

**Discussion by HernanRamirezR** [pschatzmann/ESP32-A2DP#503] [Q&A] [answered]
I'm trying to use the library for a Bluetooth speaker project while using the internal DAC of the ESP32. I'm using the following code as suggested in the documentation: 

void setup() {static const i2s_config_t i2s_config = {
        .mode = (i2s_mode_t) (I2S_MODE_MASTER | I2S_MODE_TX | I2S_MODE_DAC_BUILT_IN),
        .sample_rate = 44100, // corrected by info from bluetooth
        .bits_per_sample = (i2s_bits_per_sample_t) 16, /* the DAC module will only take the 8bits from MSB */
        .channel_format = I2S_CHANNEL_FMT_RIGHT_LEFT,
        .communication_format = (i2s_comm_format_t)I2S_COMM_FORMAT_STAND_MSB,
        .intr_alloc_flags = 0, // default interrupt priority
        .dma_buf_count = 8,
        .dma_buf_len = 64,
        .use_apll = false
    };
    a2dp_sink.set_i2s_config(i2s_config);
   a2dp_sink.start("Musica");
}

void loop() {
}

However, I get this message: 

Utilizando archivo previamente compilado: C:\Users\ZA7\AppData\Local\Temp\arduino_build_548708\libraries\ESP32-A2DP-main\BluetoothA2DPCommon.cpp.o
C:\Users\ZA7\Documents\Arduino\libraries\ESP32-A2DP-main\src\BluetoothA2DPSinkQueued.cpp: In member function 'virtual size_t BluetoothA2DPSinkQueued::write_audio(const uint8_t*, size_t)':
C:\Users\ZA7\Documents\Arduino\libraries\ESP32-A2DP-main\src\BluetoothA2DPSinkQueued.cpp:102:77: error: too many arguments to function 'void vRingbufferGetInfo(RingbufHandle_t, UBaseType_t*, UBaseType_t*, UBaseType_t*, UBaseType_t*)'
         vRingbufferGetInfo(s_ringbuf_i2s, NULL, NULL, NULL, NULL, &item_size);
                                                                             ^
In file included from C:\Users\ZA7\Documents\Arduino\libraries\ESP32-A2DP-main\src\BluetoothA2DPSink.h:19:0,
                 from C:\Users\ZA7\Documents\Arduino\libraries\ESP32-A2DP-main\src\BluetoothA2DPSinkQueued.h:3,
                 from C:\Users\ZA7\Documents\Arduino\libraries\ESP32-A2DP-main\src\BluetoothA2DPSinkQueued.cpp:2:
C:\Users\ZA7\AppData\Local\Arduino15\packages\esp32\hardware\esp32\1.0.6/tools/sdk/include/esp_ringbuf/freertos/ringbuf.h:376:6: note: declared here
 void vRingbufferGetInfo(RingbufHandle_t xRingbuffer, UBaseType_t *uxFree, UBaseType_t *uxRead, UBaseType_t *uxWrite, UBaseType_t *uxItemsWaiting);
      ^
C:\Users\ZA7\Documents\Arduino\libraries\ESP32-A2DP-main\src\BluetoothA2DPSinkQueued.cpp:118:77: error: too many arguments to function 'void vRingbufferGetInfo(RingbufHandle_t, UBaseType_t*, UBaseType_t*, UBaseType_t*, UBaseType_t*)'
         vRingbufferGetInfo(s_ringbuf_i2s, NULL, NULL, NULL, NULL, &item_size);
                                                                             ^
In file included from C:\Users\ZA7\Documents\Arduino\libraries\ESP32-A2DP-main\src\BluetoothA2DPSink.h:19:0,
                 from C:\Users\ZA7\Documents\Arduino\libraries\ESP32-A2DP-main\src\BluetoothA2DPSinkQueued.h:3,
                 from C:\Users\ZA7\Documents\Arduino\libraries\ESP32-A2DP-main\src\BluetoothA2DPSinkQueued.cpp:2:
C:\Users\ZA7\AppData\Local\Arduino15\packages\esp32\hardware\esp32\1.0.6/tools/sdk/include/esp_ringbuf/freertos/ringbuf.h:376:6: note: declared here
 void vRingbufferGetInfo(RingbufHandle_t xRingbuffer, UBaseType_t *uxFree, UBaseType_t *uxRead, UBaseType_t *uxWrite, UBaseType_t *uxItemsWaiting);
      ^
Usando librería ESP32-A2DP-main con versión 1.7.3 en la carpeta: C:\Users\ZA7\Documents\Arduino\libraries\ESP32-A2DP-main 
exit status 1
Error compilando para la tarjeta ESP32 Dev Module.

(sorry for some of the message parts being in Spanish, but I think one could understand without a direct translation) 

From what I understand there's an error with the amount of arguments that vRingbufferGetInfo is receiving, nonetheless I haven't modified any of it, and thus I'm not really sure how to fix it. I'm sort of beginner with all of this and don't know where else to ask, so any help is well received.

**Comment by RPotashov** [comment]
Hello. I thought I was the only one with this error. Apparently this is a global error. 
Could this be due to using the latest version of the IDE? We will wait for the author's response.

**Reply by HernanRamirezR** [reply]
I think it has to be. I tried in an older version of the IDE and it worked just fine, so I would recommend doing that in the meanwhile.

**Reply by RPotashov** [reply, marked as answer]
Hi. I have just updated the library for ESP32 to the latest version and everything has started to compile for me. So try it.
